service/worker/archiver/activities.go

```diff
@@ -23,80 +23,74 @@ package archiver
-	deleteBlobActivityFnName    = "deleteBlobActivity"
 	deleteHistoryActivityFnName = "deleteHistoryActivity"
-	blobstoreTimeout            = 30 * time.Second
-
-	errInvalidRequest = "archival request is invalid"
-	errGetDomainByID  = "could not get domain cache entry"
-	errConstructKey   = "could not construct blob key"
-	errGetTags        = "could not get blob tags"
-	errUploadBlob     = "could not upload blob"
-	errReadBlob       = "could not read blob"
-	errEmptyBucket    = "domain is enabled for archival but bucket is not set"
-	errConstructBlob  = "failed to construct blob"
-	errDownloadBlob   = "could not download existing blob"
-	errDeleteBlob     = "could not delete existing blob"
 
 	errDeleteHistoryV1 = "failed to delete history from events_v1"
 	errDeleteHistoryV2 = "failed to delete history from events_v2"
 
-	errHistoryMutated = "history was mutated during uploading"
-
-	errActivityPanic = "cadenceInternal:Panic"
+	errActivityPanic       = "cadenceInternal:Panic"
+	errTimeoutStartToClose = "cadenceInternal:Timeout START_TO_CLOSE"
+	errTimeoutHeartbeat    = "cadenceInternal:Timeout HEARTBEAT"
 )
 
 var (
-	uploadHistoryActivityNonRetryableErrors = []string{errGetDomainByID, errConstructKey, errGetTags, errUploadBlob, errReadBlob, errEmptyBucket, errConstructBlob, errDownloadBlob, errHistoryMutated, errActivityPanic}
-	deleteBlobActivityNonRetryableErrors    = []string{errConstructKey, errGetTags, errUploadBlob, errEmptyBucket, errDeleteBlob}
+	uploadHistoryActivityNonRetryableErrors = []string{errActivityPanic, carchiver.ErrArchiveNonRetriable.Error(), errTimeoutStartToClose, errTimeoutHeartbeat}
 	deleteHistoryActivityNonRetryableErrors = []string{errDeleteHistoryV1, errDeleteHistoryV2}
 	errContextTimeout                       = errors.New("activity aborted because context timed out")
 )
 
-const (
-	uploadErrorMsg = "Archival upload attempt is giving up, possibly could retry."
-	uploadSkipMsg  = "Archival upload request is being skipped, will not retry."
-)
-
-// uploadHistoryActivity is used to upload a workflow execution history to blobstore.
-// method will retry all retryable operations until context expires.
-// archival will be skipped and no error will be returned if cluster or domain is not figured for archival.
-// an error will be returned if context timeout, request is invalid, or failed to get domain cache entry.
-// all other errors (and the error details) will be included in the uploadResult.
-// the result will also contain a list of blob keys to delete when there's an error.
-func uploadHistoryActivity(ctx context.Context, request ArchiveRequest) (result *uploadResult, err error) {
-	return NewHistoryBlobUploader().UploadHistory(ctx, &request)
+// uploadHistoryActivity is used to archive a workflow execution history.
+// method will retry all errors except timeout errors and archiver.ErrArchiveNonRetriable.
+// it's the history archiver's responsibility to retry individual operations inside the Archive() method.
```

Comment:
Lets leave line 58 out of the comment. Its totally up to the implementor if they want to retry or not - its not their responsibility to retry.